### PR TITLE
Traits update

### PR DESCRIPTION
**Pull request by AerialMantis**
* Add bulk_post, bulk_defer, bulk_async_post and bulk_async_defer to
table of combinations.
* Update synopsis with new member function traits and free function traits.
* Add new member function traits.
* Add sections to customisation points section.
* Add sections to template functions section.
* Add list and table for free function traits.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -832,15 +932,59 @@ This sub-clause contains templates that may be used to query the properties of a
...
 | `template<class T>` <br/>`struct has_post_member` | `T` has a member function named `post` that satisfies the syntactic requirements of a one-way, non-blocking execution function of single cardinality. | `T` is a complete type. |
 | `template<class T>` <br/>`struct has_defer_member` | `T` has a member function named `defer` that satisfies the syntactic requirements of a one-way, non-blocking execution function of single cardinality. | `T` is a complete type. |
-| `template<class T>` <br/>`struct has_sync_execute_member` | `T` has a member function named `sync_execute` that satisfies the syntactic requirements of a synchronous two-way execution function of single cardinality. | `T` is a complete type. |
+| `template<class T>` <br/>`struct has_sync_execute_member` | `T` has a member function named `sync_execute` that satisfies the syntactic requirements of a synchronous two-way, potentially blocking execution function of single cardinality. | `T` is a complete type. |
```
Comment (jaredhoberock) [inline README.md]: `sync_execute()` always blocks, so the introduction of "potentially blocking" doesn't make sense here, right?

File: README.md [hunk 2]
```diff
@@ -832,15 +932,59 @@ This sub-clause contains templates that may be used to query the properties of a
...
+| `template<class T>` <br/>`struct has_then_execute_member` | `T` has a member function named `then_execute` that satisfies the syntactic requirements of an asynchronous two-way, potentially blocking execution function of single cardinality. | `T` is a complete type. |
 | `template<class T>` <br/>`struct has_bulk_execute_member` | `T` has a member function named `bulk_execute` that satisfies the syntactic requirements of a one-way, potentially blocking execution function of bulk cardinality. | `T` is a complete type. |
-| `template<class T>` <br/>`struct has_bulk_sync_execute_member` | `T` has a member function named `bulk_sync_execute` that satisfies the syntactic requirements of a synchronous two-way execution function of bulk cardinality. | `T` is a complete type. |
+| `template<class T>` <br/>`struct has_bulk_post_member` | `T` has a member function named `bulk_post` that satisfies the syntactic requirements of a one-way, potentially blocking execution function of bulk cardinality. | `T` is a complete type. |
```
Comment (jaredhoberock) [inline README.md]: `post()` and `defer()` never block, so these should be non-blocking, right?

**Comment by jaredhoberock** [comment]
Thanks Gordon. I think some of the blocking guarantees in the change may require another review. It looks like some of the guarantees are wrong possibly due to a copy/paste error.

File: README.md (hunk 2 above)
Comment (AerialMantis) [inline README.md]: Yeah that was a mistake.

**Comment by jaredhoberock** [comment]
Thanks, it looks good to me. We'll also want to incorporate the free function detection traits into the customization point specifications.

**Pull request merged**